core/models/machine.py

```diff
@@ -283,7 +284,15 @@ def convert_glance_image(glance_image, provider_uuid, owner=None):
     from core.models import AtmosphereUser
     image_id = glance_image.id
     machine_name = glance_image.name
-    application_name = machine_name  # Future: application_name will partition at the 'Application Version separator'.. and pass the version_name to create_version
+    default_version_name = '1.0'  # could be configurable if controversial
+    application_name, version_name = ProviderMachine._split_cloud_name(machine_name)
+
+    metadata_version_name = glance_image.get('version_name', default_version_name)
+    if not version_name:
+        version_name = metadata_version_name
+    if metadata_version_name != version_name:
+        logger.warn("Deprecation warning: Using version_name:%s but metadata defines 'application_version=%s'",
+                    version_name, metadata_version_name)
```

Comment:
I would change this logic like so:
```
version_name = glance_image.get('version_name', default_version_name)
```
Make the metadata the source of truth, not something in the image name.

Reply:
on further inspection what i'm advocating for already existed